src/radical/pilot/session.py

```diff
@@ -968,3 +985,29 @@ def excuse():
 
 
 # ------------------------------------------------------------------------------
+
+
+@dataclasses.dataclass(frozen=True)
+class _CloseOptions:
+    """Options and validation for Session.close().
+
+    **Arguments:**
+        * **cleanup**   (`bool`):
+          Remove session from MongoDB (implies * terminate). (default False)
+        * **terminate** (`bool`):
+          Shut down all pilots associated with the session. (default True)
+        * **download** (`bool`):
+          Fetch pilot profiles and database entries. (default False)
+
+    """
+    cleanup: bool = False
+    # Warning: Previously, passing `cleanup=None` would reverse the default and
```

Comment:
Was this logic intentional? Is it okay to remove the distinction between unspecified *cleanup* and explicit *cleanup=None*? If not, the field is really an explicit Union rather than an implicit Optional, and it might be more appropriate to use either a TypedDict or just a simple dict without encapsulated logic.

Reply:
resolved (if user will set `s.close(cleanup=None)` then it will be interpreted as `not True`, but only `True` values are targeted)